src/runtime/classmanager.cs

```diff
@@ -76,32 +71,32 @@ private static int OnVisit(IntPtr ob, IntPtr arg)
             var clrObj = ManagedType.GetManagedObject(ob);
             if (clrObj != null)
             {
-                clrObj.CallTypeTraverse(OnVisit, arg);
+                clrObj.CallTypeTraverse(TraverseTypeClear, arg);
                 clrObj.CallTypeClear();
             }
             return 0;
         }
 
-        internal static void StashPush(RuntimeDataStorage storage)
+        internal static void SaveRuntimeData(RuntimeDataStorage storage)
         {
             var contexts = storage.AddValue("contexts",
-                new Dictionary<IntPtr, PyObjectSerializeContext>());
+                new Dictionary<IntPtr, InterDomainContext>());
             storage.AddValue("cache", cache);
             foreach (var cls in cache.Values)
             {
                 // This incref is for cache to hold the cls,
-                // thus no need for decreasing it at StashPop.
+                // thus no need for decreasing it at RestoreRuntimeData.
```

Comment:
Does this mean whatever is stored in `storage` leaks every domain restart?

Reply:
If I understand the code correctly, at shutdown, the types are cleared. To avoid them actually going away, we incref them. Why doesn't this leak? Because this incref is equivalent to the new reference received when creating the type (except is is never created when reloading a domain, only preserved)
@amos402  is this correct?

Reply:
Correct. The types' data will be recovering at the next startup.
Yet there's still not having any error fallback now. The reload mode is not a completed mode right now.